common-app-lib/src/cmr/common_app/services/search/params.clj

```diff
@@ -215,6 +216,55 @@
   [concept-type params]
   (default-parse-query-level-params concept-type params))
 
+(defn- generate-conditions
+  "Converts given parameters into elasticsearch conditions"
+  [context concept-type params options]
+  (map (fn [[param value]]
+           (parameter->condition context concept-type param value options))
+        params))
+
+(defn- create-combined-conditions
+  "When valid spatial parameters are given and the spatial OR option is passed,
+   create proper search conditions"
+  [context concept-type options spatial-params non-spatial-params]
+  (let [spatial-conditions (when-not (empty? spatial-params)
+                       (generate-conditions
+                        context concept-type spatial-params options))
+        combined-spatial-conditions (gc/or-conds spatial-conditions)
+        non-spatial-conditions (when-not (empty? non-spatial-params)
+                                 (generate-conditions
+                                  context concept-type non-spatial-params options))
+        combined-non-spatial-conditions (when non-spatial-conditions
+                                          (gc/and-conds non-spatial-conditions))]
+    (if (and combined-spatial-conditions combined-non-spatial-conditions)
+        (gc/and-conds (conj combined-spatial-conditions combined-non-spatial-conditions))
+        (conj combined-spatial-conditions combined-non-spatial-conditions))))
+
+(defn- combine-query-conditions
+  "Appropriately combine query operators based on given options"
+  [context concept-type options params query-attribs]
+  (let [valid-spatial-params [:bounding-box :circle :point :line :polygon :bounding-box]
+        spatial-option (if (= (get-in options [:spatial :or]) "true")
+                           :or
+                           :and)
+        user-spatial-params (when-not (empty? params)
+                              (select-keys params valid-spatial-params))
+        user-non-spatial-params (-> params
+                                    (data/diff user-spatial-params)
+                                    first)
+        conditions (generate-conditions context concept-type params options)]
```

Comment:
It is a waste to build the conditions here as when spatial options is `OR`, we don't use the conditions that is build here.